.github/workflows/build-and-runtime-test.yml

```diff
@@ -101,6 +102,15 @@ jobs:
           USERNAME: ${{ secrets.USERNAME }}
           NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
           VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
+      - name: Upload videos and screenshots
+        uses: actions/upload-artifact@v3
+        if: ${{ failure() && steps.e2e.outcome != 'success' }}
+        with:
+          name: canary-cypress-error-{{ matrix.path }}
```

Comment:
Appended `${{ matrix.path }}` so that uploaded zip file names are unique and do not overwrite each other.